morpheus/utils/module_utils.py

```diff
@@ -179,50 +180,41 @@ def merge_dictionaries(primary_dict, secondary_dict):
     "D": "%Y-%m-%d",
     "W": "%Y-%U",
     "M": "%Y-%m",
-    "Y": "%Y",
-    "Q": "%Y-%q",
-    "A": "%Y"
+    "Y": "%Y"
 }
 
 
-def to_period_cudf_approximation(df, period):
+def to_period_approximation(data_df: typing.Union[cudf.DataFrame, pd.DataFrame], period: str):
```

Comment:
Look at this PR: https://github.com/nv-morpheus/Morpheus/pull/912 for using the `resample` function to simplify this